### ISSUE TITLE
Make Workbench use Drupal's built in REST interfaces for files and media

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#171]
... instead of using Islandora's MediaSourceController. As discussed at the 2020-12-09 Islandora 8 Tech call re. https://github.com/Islandora/documentation/issues/1481, this controller provides a lot of Islandora-specific plumbing that it would be disruptive and difficult to refactor. It is likely to be an order of magnitude simpler to adapt Workbench's `create_media()` function to use Drupal's File and Media REST resources to attach a file and then a media to a node.

**Comment by mjordan** [comment]
Probably wait until Drupal 9.1 to implement this, since creating media via REST suffered from https://www.drupal.org/project/drupal/issues/3072076. After that is fixed, the following should work:

```
curl -v -X POST -uadmin:islandora -H'Accept: application/vnd.api+json' -H'Content-Type: application/octet-stream' -H'Content-Disposition: file; filename="test.jpg"' --data-binary @bcp3356.jpg  "http://localhost:8000/jsonapi/media/image/field_media_image"
```
then PATCH with field data like:

```
 curl -v -X PATCH -uadmin:islandora -H'Accept: application/vnd.api+json' -H'Content-Type:application/vnd.api+json' --data '{"data":{"type": "media--image", "id":"f6b36617-3982-4746-b5cc-8e14f60519d4","attributes":{"name":"My New Name"}}}' "http://localhost:8000/jsonapi/media/image/f6b36617-3982-4746-b5cc-8e14f60519d4"
```

**Comment by mjordan** [comment]
OK, back in business. In Drupal 9.2.0-dev, the issue linked above is resolved.

This now works for creating file:

curl -v -X POST -uadmin:islandora -H'Accept: application/vnd.api+json' -H'Content-Type: application/octet-stream' -H'Content-Disposition: file; filename="test.jpg"' --data-binary @bcp3356.jpg  "http://localhost:8000/jsonapi/media/image/field_media_image"

Then, for creating the file's media:

curl -v -X POST -uadmin:islandora -H'Accept: application/vnd.api+json' -H'Content-type: application/vnd.api+json' --data @media.json  "http://localhost:8000/jsonapi/media/image"

with the following as the `--data @media.json`:

```json
{
  "data":{
     "type":"media--image",
     "attributes":{
        "name":"REST-create media"
     },
     "relationships":{
        "field_media_of":{
           "data":{
              "type":"node--islandora_object",
              "id":"fa12d7ce-f66c-494f-b149-d95c3c201210"
           }
        },
        "field_media_image":{
           "data":{
              "type":"file--file",
              "id":"5060c33a-789f-4850-a2f9-2f6e05e2dd65"
           }
        },
        "field_media_use":{
           "data":{
              "type":"taxonomy_term--islandora_media_use",
              "id":"cc5e6f00-be2f-4d07-b047-be39cd9b7387"
           }
        }
     }
  }
}
```

**Comment by mjordan** [comment]
Some preliminary tasks:

First, to create files, Workbench will need a  media type -> media field mapping like this:

```
'file': 'field_media_file',
'document': 'field_media_document',
'image': 'field_media_image',
'audio': 'field_media_audio_file',
'video': 'field_media_video_file'
```

and this will need to be extensible via configuration, to accommodate custom media types.

Second, to create media, we'll need to be able to get UUIDs for instances of nodes, files, and taxo terms, as illustrated abobe.

**Comment by mjordan** [comment]
`config['media_fields']` option added in 8d1cf750eec47510dc9030f38f9ea6485399b4a6. If present in config file, allows adding/overriding which field is used for a media bundle's file.

**Comment by mjordan** [comment]
`get_term_uuid(config, term_id)` added in 2c5243480f2b74df34e3f0b8a9c618ae3f735195.

**Comment by mjordan** [comment]
#254 adds code to check version of Drupal core. Depends on islandora_workbench_integration commit 638d172b2589463c36b2032c6400f62063fc932d. To put this into production, uncomment `# check_drupal_core_version(config)` in workbench_utils.py.

**Comment by mjordan** [comment]
Noteworthy changes to Workbench resulting from development on this issue:

* Workbench now polls Workbench Integration for Drupal's version number.
    * If Drupal's version is less than 9.2, Workbench uses Islandora's MediaSourceController to create media (i.e., what it's been doing up until now). 
    * If Drupal's version is equal to or greater than 9.2, Workbench uses Drupal's core REST endpoints to create files and media. No dependency on JSON:API.
    * Note: Users must pull in the most recent changes to the `main` branch in https://github.com/mjordan/islandora_workbench_integration. Otherwise, Workbench will issue an error about not being able to detect Drupal's version and exit.
* Image media get assigned an alt text value as specified in the CSV `image_alt_text` column. If that column is not present, or is empty, the parent node's `title` value is used as the image's alt text.
* Resolves media filename bug reported in https://github.com/mjordan/islandora_workbench/issues/251.
* Resolves https://github.com/mjordan/islandora_workbench/issues/77.

**Comment by mjordan** [comment]
Merged into main with b37f3539a17559f9a68247114972b92543fbd376.
